### PR TITLE
CUST-107 bigquery pagination discovery

### PR DESCRIPTION
**Pull request by RohitJ1995**

### CODE REVIEW

File: airbyte-integrations/connectors/source-bigquery/src/main/java/io/airbyte/integrations/source/bigquery/BigQuerySource.java [hunk 1]
```diff
@@ -35,10 +36,15 @@
...
+
+import io.airbyte.protocol.models.v0.AirbyteMessage;
+import io.airbyte.protocol.models.v0.ConfiguredAirbyteCatalog;
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
 
-public class BigQuerySource extends ExtendAbstractDBSource<StandardSQLTypeName, BigQueryDatabase> implements Source {
+public class BigQuerySource extends AbstractDbSource<StandardSQLTypeName, BigQueryDatabase> implements Source {
+
+  private ConfiguredAirbyteCatalog catalogConfig;
```
Comment (jhecking) [inline airbyte-integrations/connectors/source-bigquery/src/main/java/io/airbyte/integrations/source/bigquery/BigQuerySource.java:47]: As discussed previously, can you add a comment to explain the usage of this variable and the dependency between the `read` and `discoverInternal` functions that this is solving?

**Review by jhecking — COMMENTED** [review]
Looks good to me. Needs to be moved to `airbyte-private` repo, though.

I assume the `querySampleDataStream` functionality has been shifted to some other PR? Or why did it get removed here?

And what's the difference between the `ExtendedAbstractDBSource` and `AbstractDbSource`? Doesn't seem like this change is related to the pagination logic?

**Comment by RohitJ1995** [comment]
> I assume the querySampleDataStream functionality has been shifted to some other PR? Or why did it get removed here?
it's overridden in BorneoBigQuerySource.java with the complete read implementation

> And what's the difference between the ExtendedAbstractDBSource and AbstractDbSource? 
removed the entire ExtendAbstractDBSource as we dint wanted to change make many changes to airbyte classes

> Doesn't seem like this change is related to the pagination logic? 
All these were a part of the unified connector framework changes, have not merged it the main branch (borneo-airbyte)
have asked @niteshy  to review these changes also in a separate PR: 
https://github.com/Securezapp/airbyte/pull/13/files

File: airbyte-integrations/connectors/source-bigquery/src/main/java/io/airbyte/integrations/source/bigquery/BigQuerySource.java (hunk 1 above)
Comment (RohitJ1995) [reply airbyte-integrations/connectors/source-bigquery/src/main/java/io/airbyte/integrations/source/bigquery/BigQuerySource.java:51]: done

**Comment by ashwin-borneo** [comment]
Looks good to me.

**Review by ashwin-borneo — APPROVED** [review]

**Pull request merged**